### PR TITLE
KGDataset.from_dataframe + custom data notebook

### PR DESCRIPTION
**Pull request by AlCatt91**
Added an ease-of-use method to build a `KGDataset` taking in directly labelled triples (organized in a pandas dataframe), performing the label -> ID conversion behind the curtain and re-ordering, if needed, to cluster entities of the same type. It is meant to be even higher-level than the `from_triples` method.
Using this, I also wrote up a notebook, as we were planning to do, to show how to build a custom dataset, using as example the [OpenBioLink](https://github.com/openbiolink/openbiolink) dataset (there are the usual concerns about licensing of data sources, but I hope that, if we use it just for this small demo notebook, it shouldn't be a problem?) Any feedback is welcome!

### CODE REVIEW

**Comment by AlCatt91** [comment]
I added to this PR with a few updates:
- Added GC disclaimer for use of datasets (see README and NOTICE);
- Added list of available datasets in README;
- Updated CONTRIBUTING with details on how to work on a fork (preferred way to submit PRs);
- Some updated to dataset.py, making use of the `from_dataframe` builder to cut down on code;
- Added dataloader for OpenBioLink.

The updates to the notebook are from black formatting (can disregard).

**Review by danjust — COMMENTED** [review]
LGTM - see comment regarding OpenBioLink notebook

**Comment by AlCatt91** [comment]
Thanks! I changed the dataset in the new notebook to biokg, please have a final look when you can :)

File: notebooks/0_custom_KG_dataset.ipynb [hunk 1]
```diff
@@ -98,76 +97,143 @@
             "source": [
                 "## The dataset\n",
                 "\n",
-                "We download the directed, high-quality version of [OpenBioLink2020](https://github.com/openbiolink/openbiolink#benchmark-dataset) directly from the link provided by the authors. This shouldn't take more than a minute.\n",
-                "\n",
-                "Notice that OpenBioLink2020 integrates data from other sources, whose licensing terms are detailed in [this table](https://openbiolink.readthedocs.io/en/latest/sources.html) and should be minded when utilizing or redistributing the dataset files."
+                "We download the OGBL-BioKG Knoweldge Graph using the `ogb` package (see [here](https://ogb.stanford.edu/docs/linkprop/#data-loader) for details on how to use it). This shouldn't take more than a couple of minutes."
```
Comment (danjust) [inline notebooks/0_custom_KG_dataset.ipynb:100]: typo in Knowledge Graph

**Review by danjust — COMMENTED** [review]
Looks great - just spotted one typo

File: notebooks/0_custom_KG_dataset.ipynb (hunk 1 above)
Comment (AlCatt91) [reply notebooks/0_custom_KG_dataset.ipynb]: Thanks, fixed!

**Review by danjust — APPROVED** [review]

**Pull request merged**